### PR TITLE
Fixes 115: Added blocking keyboard player

### PR DESCRIPTION
**Pull request by pathway**
"Blocking" means the whole game waits for user input.

Test this with:

pom_battle --agents=playerblock::arrows,test::agents.SimpleAgent,test::agents.SimpleAgent,test::agents.SimpleAgent --config=PommeFFACompetition-v0 --render

pom_battle --agents=playerblock::wasd,test::agents.SimpleAgent,test::agents.SimpleAgent,test::agents.SimpleAgent --config=PommeFFACompetition-v0 --render

You can still use the original player (non-blocking):

pom_battle --agents=player::arrows,test::agents.SimpleAgent,test::agents.SimpleAgent,test::agents.SimpleAgent --config=PommeFFACompetition-v0 --render

### CODE REVIEW

File: pommerman/agents/player_agent_blocking.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+from time import time
+import click
```
Comment (cinjon) [inline pommerman/agents/player_agent_blocking.py]: I am unfamiliar with this package. Is it https://click.palletsprojects.com/en/7.x/? 
If so, that page suggests it's for CLIs. Why not do as we did here with Pyglet --> https://github.com/MultiAgentLearning/playground/blob/master/pommerman/agents/player_agent.py

File: pommerman/agents/player_agent_blocking.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+from .. import characters
+from .. import constants
+
+
+class PlayerAgentBlocking(BaseAgent):
+    """Block for keyboard input."""
+
+    def __init__(self, character=characters.Bomber, agent_control='arrows'):
+        super(PlayerAgentBlocking, self).__init__(character)
+        self._character.agent_image = 'agent-player'
```
Comment (cinjon) [inline pommerman/agents/player_agent_blocking.py]: Why are you setting this?

File: pommerman/agents/player_agent_blocking.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    def act(self, obs, action_space):
+        key = click.getchar()
+        #key = click.pause()
+        #print(key,repr(key))
+        if self.agent_control=='arrows':
+            if key=='\x1b[C': return constants.Action.Right.value
+            if key=='\x1b[D': return constants.Action.Left.value
+            if key=='\x1b[A': return constants.Action.Up.value
+            if key=='\x1b[B': return constants.Action.Down.value
```
Comment (cinjon) [inline pommerman/agents/player_agent_blocking.py]: Make these more user-friendly please, e.g.
https://github.com/MultiAgentLearning/playground/blob/master/pommerman/agents/player_agent.py#L80.

File: pommerman/agents/player_agent_blocking.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    def __init__(self, character=characters.Bomber, agent_control='arrows'):
+        super(PlayerAgentBlocking, self).__init__(character)
+        self._character.agent_image = 'agent-player'
+        self.agent_control=agent_control
+
+    def act(self, obs, action_space):
+        key = click.getchar()
+        #key = click.pause()
+        #print(key,repr(key))
```
Comment (cinjon) [inline pommerman/agents/player_agent_blocking.py]: remove

File: pommerman/agents/player_agent_blocking.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+class PlayerAgentBlocking(BaseAgent):
+    """Block for keyboard input."""
+
+    def __init__(self, character=characters.Bomber, agent_control='arrows'):
+        super(PlayerAgentBlocking, self).__init__(character)
+        self._character.agent_image = 'agent-player'
+        self.agent_control=agent_control
+
+    def act(self, obs, action_space):
+        key = click.getchar()
```
Comment (cinjon) [inline pommerman/agents/player_agent_blocking.py]: Does this block? Is that what the mechanism is?

File: pommerman/agents/player_agent_blocking.py (hunk 2 above)
Comment (pathway) [reply pommerman/agents/player_agent_blocking.py]: @cinjon oops its from a branch I made with customizable icons.  It should not be in this commit, can remove.

File: pommerman/agents/player_agent_blocking.py (hunk 5 above)
Comment (pathway) [reply pommerman/agents/player_agent_blocking.py]: yes.

File: pommerman/agents/player_agent_blocking.py (hunk 1 above)
Comment (pathway) [reply pommerman/agents/player_agent_blocking.py]: Yes, that click.
I just wanted simple code, I found it simpler this way.
I understand if you dont like this dependency, Im sure there are other ways.

Comment (pathway) [reply pommerman/agents/player_agent_blocking.py]: It does cli things but also some terminal handling (like getchar)

File: pommerman/agents/player_agent_blocking.py (hunk 2 above)
Comment (pathway) [reply pommerman/agents/player_agent_blocking.py]: I removed

**Comment by pathway** [comment]
Still have concerns here @cinjon ?

File: pommerman/agents/player_agent_blocking.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        super(PlayerAgentBlocking, self).__init__(character)
+        self.agent_control=agent_control
+
+    def act(self, obs, action_space):
+        key = click.getchar()
+        if self.agent_control=='arrows':
+            if key==K_PREFIX+K_RT: return constants.Action.Right.value
+            if key==K_PREFIX+K_LF: return constants.Action.Left.value
+            if key==K_PREFIX+K_UP: return constants.Action.Up.value
+            if key==K_PREFIX+K_DN: return constants.Action.Down.value
```
Comment (cinjon) [inline pommerman/agents/player_agent_blocking.py]: nit: please use an enum or make it explicit. it's much nicer and more legible to have it read `k.UP`, `k.RIGHT`, ...

**Comment by cinjon** [comment]
seems good. i'd appreciate it if you ran it through `yapf -i --style google` quickly before submitting.

**Comment by cinjon** [comment]
also, please squash the commits together.

**Comment by pathway** [comment]
Superseded by https://github.com/MultiAgentLearning/playground/pull/208

**Comment by cinjon** [comment]
we can close this, ja?

**Pull request closed without merging**